src/client.ts

```diff
@@ -175,26 +176,14 @@ export class OpenFeatureClient implements Client {
       flagValueType: flagType,
       clientMetadata: this.metadata,
       providerMetadata: OpenFeature.providerMetadata,
-      context: globalAndClientContext,
+      context: mergedContext,
     };
 
     try {
-      const mergedHookContext = await this.beforeHooks(allHooks, hookContext, options);
-
-      // merge context in order: global, client, hook, invocation
-      const mergedContext = {
-        ...mergedHookContext,
-        ...invocationContext,
-      };
-
-      // if a transformer is defined, run it to prepare the context
-      const transformedContext =
-        typeof this.provider.contextTransformer === 'function'
-          ? await this.provider.contextTransformer(mergedContext)
-          : mergedContext;
+      const frozenContext = await this.beforeHooks(allHooks, hookContext, options);
```

Comment:
chose this name because the context object is actually frozen (immutable here) after the before hooks have run...